gcc/rust/ast/rust-ast-dump.cc

```diff
@@ -951,49 +945,24 @@ Dump::visit (StructStruct &struct_item)
 {
   stream << "struct " << struct_item.get_identifier ();
   if (struct_item.has_generics ())
-    emit_generic_params (struct_item.get_generic_params ());
+    visit (struct_item.get_generic_params ());
 
   // FIXME: where-clause
 
-  stream << " {";
-
-  auto &fields = struct_item.get_fields ();
-
-  indentation.increment ();
-  for (auto &field : fields)
-    {
-      stream << '\n' << indentation;
-      format_struct_field (field);
-      stream << ',';
-    }
-  indentation.decrement ();
-
-  if (fields.size () > 0)
-    stream << '\n' << indentation;
-  stream << "}\n";
+  visit_items_as_block (struct_item.get_fields (), '{', '}');
```

Comment:
One thing to be careful of, which I can't remember, is whether or not our AST distinguishes between
```rust
struct Foo;
```
and 
```rust
struct Foo {}
```
as they are quite different and should be dumped differently

Reply:
I looked into the parser and yes, that case should be handled. There is a bool `is_unit` which determined it. But is in not related to this refactor - it was not handled before, so I will open a new PR.

Reply:
Solved by #1625